modules/azure/src/main/java/org/testcontainers/containers/KeyStoreUtils.java

```diff
@@ -0,0 +1,65 @@
+package org.testcontainers.containers;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+
+import static java.nio.file.Files.readAllBytes;
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+import static org.testcontainers.containers.Constants.EMULATOR_CERTIFICATE_ALIAS;
+import static org.testcontainers.containers.Constants.EMULATOR_CERTIFICATE_ENDPOINT_URI;
+import static org.testcontainers.containers.Constants.STORE_PASSWORD;
+import static org.testcontainers.containers.Constants.STORE_TYPE;
+
+/**
+ * @author Onur Kagan Ozcan
+ */
+final class KeyStoreUtils {
+
+    static void downloadPemFromEmulator(String endpoint, Path pemResourceOutput) {
```

Comment:
Is there a location inside the container filesystem, at which the certificate is stored? If yes, we can remove `URLSSLCertificateCheck` completely and just use `copyFileFromContainer`. I would be much more comfortable with this.

Reply:
👍 Added to my patch list.

wow, I didn't know that, this will simplify too much :)